streams/src/main/java/org/apache/kafka/streams/kstream/internals/NamedInternal.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.kstream.Named;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class NamedInternal extends Named {
+
+    private static final NamedInternal EMPTY = new NamedInternal();
+
+    public static NamedInternal empty() {
+        return EMPTY;
+    }
+
+    public static NamedInternal with(final String name) {
+        return new NamedInternal(name);
+    }
+
+    /**
+     * Creates a new {@link NamedInternal} instance.
+     *
+     * @param internal  the internal name.
+     */
+    public NamedInternal(final Named internal) {
+        super(internal);
+    }
+
+    public NamedInternal(final String internal) {
+        super(internal);
+    }
+
+    private NamedInternal() {
```

Comment:
Why do we need this?

Reply:
The empty constructor is used to build an empty `NamedInternal`, see `NamedInternal.empty()`